lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1038,12 +1093,34 @@ mkUnsignedTx era ttl cs md wdrls certs fees =
         , Cardano.txUpdateProposal =
             Cardano.TxUpdateProposalNone
 
-        , Cardano.txMintValue =
-            Cardano.TxMintNone
+        , Cardano.txMintValue = 
+            let
+              forgeOutAmt :: Cardano.TxOut Cardano.MaryEra -> Cardano.Value
+              forgeOutAmt (Cardano.TxOut _addr (Cardano.TxOutAdaOnly _ _)) = mempty
+              forgeOutAmt (Cardano.TxOut _addr (Cardano.TxOutValue _ v))   = v
+            in
+              case foldMap forgeOutAmt forgeOuts of
+                amt | amt == mempty -> Cardano.TxMintNone
+                amt -> Cardano.TxMintValue Cardano.MultiAssetInMaryEra amt
         }
       where
         toErrMkTx = ErrConstructedInvalidTx . T.pack . Cardano.displayError
 
+        toCardanoValue :: TokenMap -> Cardano.Value
+        toCardanoValue tokenMap =
+          Cardano.valueFromList $ Bifunctor.bimap (\(AssetId pol name) -> mkAssetId pol name) (fromIntegral . unTokenQuantity) <$> TokenMap.toFlatList tokenMap
```

Comment:
I'd also suggest using `<&>` here, or at least breaking this line up in some other way.